### ISSUE TITLE
tf.keras implementation

### ISSUE THREAD

**Issue by qzhu2017** [sayred1/tf.keras_GCN#3]
Todo
- 1, add functions for `attention GCN` and `gated GCN`
- 2, check if the normalization helps.
- 3, investigate how embedding works in `tf.keras`.

Done
- 1, analyze the evolution of weights/bias. (by adding `histogram_freq` in `keras.callbacks.TensorBoard`)

**Comment by qzhu2017** [comment]
After a survey on the embedding techniques. I believe it is much easier to do it with scikit-learn, instead of Tensorflow. Let's forget about this idea for now.

**Comment by qzhu2017** [comment]
@sayred1 I suggest you add the attention GCN and gated GCN now. I may spend some time on other things.

**Comment by sayred1** [comment]
@qzhu2017 Will do, I just checked whether or not normalizing helps with results, it seems to not. Each run seemed to converge to nearly the same values. I uploaded two quick plots of MSE and MAE training comparisons for four runs (two normalized, two unnormalized), two layers, 1000 epochs, 100 batches, over 1000 molecules.

I'll go ahead and add the rest of the methods into your code.

**Comment by sayred1** [comment]
Hey, @qzhu2017 If you don’t mind, I’d like to add GCN gate and attention tomorrow. I currently have two assignments due within the next 24 hours.

**Comment by sayred1** [comment]
@qzhu2017 I've finished adding the other models into your code and I am currently running some tests before uploading to the repo.

Also, I forgot to mention that I will be going out of town in a few hours from now until Thursday. If I can't get the testing done soon, I will upload the script to the repo.
